test/passes/translate-to-fuzz_all-features.txt

```diff
@@ -296,435 +291,10 @@
-               )
-               (v128.const i32x4 0x10000000 0x00000000 0x00000080 0x00000000)
-              )
-             )
-             (br $label$1)
-            )
-           )
-          )
-         )
-        )
-        (return
-         (local.get $4)
-        )
-       )
-      )
-     )
-     (nop)
-    )
-   )
    (nop)
    (local.get $4)
   )
  )
- (func $func_9 (param $0 f64) (result nullref)
-  (local $1 nullref)
-  (local $2 f32)
-  (local $3 anyref)
-  (local $4 exnref)
-  (block
-   (if
-    (i32.eqz
-     (global.get $hangLimit)
-    )
-    (return
-     (ref.null)
-    )
-   )
-   (global.set $hangLimit
-    (i32.sub
-     (global.get $hangLimit)
-     (i32.const 1)
-    )
-   )
-  )
-  (local.get $1)
- )
- (func $func_10 (param $0 f32) (result f64)
-  (local $1 anyref)
-  (local $2 v128)
-  (local $3 i32)
-  (local $4 v128)
-  (local $5 funcref)
-  (local $6 f64)
-  (local $7 v128)
-  (local $8 f32)
-  (local $9 nullref)
-  (local $10 exnref)
-  (local $11 exnref)
-  (local $12 anyref)
-  (local $13 nullref)
-  (local $14 (v128 i32 exnref v128))
-  (local $15 i64)
-  (block
-   (if
-    (i32.eqz
-     (global.get $hangLimit)
-    )
-    (return
-     (local.get $6)
-    )
-   )
-   (global.set $hangLimit
-    (i32.sub
-     (global.get $hangLimit)
-     (i32.const 1)
-    )
-   )
-  )
-  (select
-   (local.get $6)
-   (local.get $6)
-   (if (result i32)
-    (i32.eqz
-     (local.get $3)
-    )
-    (i32.const -2147483648)
-    (local.get $3)
-   )
-  )
- )
- (func $func_11 (result i64 v128 f32 v128 v128)
-  (local $0 (v128 f32 f64 f64 f32))
-  (block
-   (if
-    (i32.eqz
-     (global.get $hangLimit)
-    )
-    (return
-     (tuple.make
-      (i64.const 1250326222040357399)
-      (v128.const i32x4 0x616a003f 0x8000ff02 0xffa20040 0x7fff5750)
-      (f32.const -3402823466385288598117041e14)
-      (v128.const i32x4 0x0400ff81 0x16168000 0x04000004 0x00000502)
-      (v128.const i32x4 0x010e0007 0x04020e0f 0xc7000000 0xcf000000)
-     )
-    )
-   )
-   (global.set $hangLimit
-    (i32.sub
-     (global.get $hangLimit)
-     (i32.const 1)
-    )
-   )
-  )
-  (block $label$0
-   (nop)
-   (return
-    (tuple.make
-     (i64.const -20)
-     (v128.const i32x4 0x7fffffff 0x00000000 0x00000000 0xffffff80)
-     (f32.const -562949953421312)
-     (v128.const i32x4 0x7fffffff 0x00000000 0x00000000 0xffffff80)
-     (v128.const i32x4 0x0aff170d 0x490b3100 0x01800055 0x20004200)
-    )
-   )
-  )
- )
- (func $func_12 (param $0 i64) (param $1 v128) (result f32)
-  (local $2 (exnref i32 nullref exnref nullref f64))
-  (local $3 i32)
-  (local $4 i64)
-  (local $5 f32)
-  (local $6 i64)
-  (local $7 f32)
-  (local $8 nullref)
-  (local $9 f64)
-  (block
-   (if
-    (i32.eqz
-     (global.get $hangLimit)
-    )
-    (return
-     (f32.const 57)
-    )
-   )
-   (global.set $hangLimit
-    (i32.sub
-     (global.get $hangLimit)
-     (i32.const 1)
-    )
-   )
-  )
-  (return
-   (f32.const 6.254571977675003e-36)
-  )
- )
```

Comment:
it's somewhat unfortunate that the test output here becomes almost uninteresting (the one user function, `func_8`, is almost empty) but not sure it's worth changing the test input for.

maybe it would be a good idea (separate from this PR) to remove this testcase, and instead add a python unit test that runs the fuzzer on various things and runs `--metrics` to see what nodes are emitted. We could verify that way that a `try` etc. is emitted by the fuzz code, which isn't tested atm.

Reply:
- What kinds of various inputs do you think would be good candidates?
- What's the advantage of those tests being Python unit tests? I'm asking because, I thought the test would be something like running the fuzzer on a fixed set of input and compare its metrics with the pregenerated correct output (such as .txt file in our normal tests. Is the comparison easier in Python unit tests?

Reply:
The inputs could be a set of 10 big random binary files. Enough that with probably almost 1 we will see every node type being generated.

I think a python unit test would be easier, since then we can just do an `if` to check each node is there. We don't care about the number of nodes or other details, just that the fuzzer emits all types. With the current `.txt` files, when we update them, it would be easy to miss if a node type stops being generated for example (the metrics output would go from 20 lines to 19).

Reply:
I checked the generated code, and funnily, before `recombine()`, the code was a lot longer and contains exception handling instructions, but after `recombine()` it becomes this, which is I think possible? But unfortunate...

I don't think you'd want to have those 10 big random binary files to be uploaded I guess? Do you mean they can be generated on the fly?

Reply:
I didn't think that far, but one option might be to upload them, another to use a deterministic random number generator.

Reply:
I'll try that as a followup patch later.

Reply:
FYI after https://github.com/WebAssembly/binaryen/pull/2880/commits/c30e53559698ae02514a09642f673cf89631d98c, the result of this fuzz case became longer again. The unit tests are good to have anyway though.

Reply:
Sorry, nevermind the comment I posted here, I ran the tests without building the new code...